### PR TITLE
Update irradiance.haydavies docs; use Wm⁻² instead of W/m^2 throughout `pvlib.irradiance`

### LINKED ISSUE #2183: Missing Descriptions for Formula Components in `haydavies`
The formula components for the [`haydavies`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.haydavies.html#pvlib.irradiance.haydavies) function seems to not be properly defined. The current formula is:

$I_{d} = DHI \left( A R_b + (1 - A) \left(\frac{1 + \cos \beta}{2}\right) \right)$

However, the documentation does not provide definitions or explanations for the components $A$ and $R_b$. While $DHI$ and $\beta$ are relatively straightforward, I feel it would be beneficial to include their descriptions for completeness. 

Additionally, [`isotropic`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.isotropic.html), [`klutcher`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.klucher.html) and [`reindl`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.reindl.html)   also lack description of formula components.

#### Request

1. Update the documentation for the [`haydavies`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.haydavies.html#pvlib.irradiance.haydavies) function to include descriptions for the following components:
   - $A$
   - $R_b$
   - $DHI$
   - $\beta$

2. Update the documentation for the [`isotropic`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.isotropic.html), [`klutcher`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.klucher.html) and [`reindl`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.reindl.html) (shall different issues be created for each function?)

I think that providing these definitions will improve clarity.

Thank you!

### PR DESCRIPTION
**Pull request by RDaxini**
- [X] Partially closes #2183
 - [X] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [X] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

**Comment by RDaxini** [comment]
Is anyone aware of an active (permanent?) link to the original Hay and Davies publication? I think it would help to link that into the references but I have been unable to find a suitable link yet.

**Comment by RDaxini** [comment]
[87fc372](https://github.com/pvlib/pvlib-python/pull/2191/commits/87fc372f52316e42a787b49613322781ff63f42e) is my personal stylistic preference for typesetting the units of watt per metre squared. This can be reverted if people disagree.

**Comment by kandersolar** [comment]
> Is anyone aware of an active (permanent?) link to the original Hay and Davies publication? I think it would help to link that into the references but I have been unable to find a suitable link yet.

It's available at archive.org, although you have to create an account and "check it out": https://archive.org/details/proceedingsfirst00cana/mode/2up

I do not know if a more convenient alternative is available.

**Comment by RDaxini** [comment]
> It's available at archive.org, although you have to create an account and "check it out": https://archive.org/details/proceedingsfirst00cana/mode/2up
> 
> I do not know if a more convenient alternative is available.


I have added that one in for now ([1063804](https://github.com/pvlib/pvlib-python/pull/2191/commits/10638048b8622c36eb97d94005d7f6cd93d389bf))
I'll add an up to date "last accessed" just before this PR is merged

File: pvlib/irradiance.py
Comment (cwhanse) [inline pvlib/irradiance.py:812]: Why are we dropping [1]? It was used here because it is easy to find (the 1980 Hay/Davies paper is not) and it contains a concise description of the Hay/Davies model.

**Review by cwhanse — COMMENTED** [review]
Concur with the additional description of terms in the Hay/Davies model equation. I disagree with changing the primary reference.

File: pvlib/irradiance.py
Comment (RDaxini) [reply pvlib/irradiance.py:812]: Ah okay I understand why it's there now. Since the [1] was not attributed anywhere in the docstring, I removed it. I have added it back in with an in-text reference ([3a9b965](https://github.com/pvlib/pvlib-python/pull/2191/commits/3a9b9659b5a241071a1dbdc10e878c887a796807)). Thanks for the suggestion

**Review by kandersolar — COMMENTED** [review]
I'm neutral on the units formatting changes.  The new form is a little nicer to look at, but I foresee myself grumbling for having to copy/paste that superscript 2 in future code edits :P

**Comment by RDaxini** [comment]
> I'm neutral on the units formatting changes. The new form is a little nicer to look at, but I foresee myself grumbling for having to copy/paste that superscript 2 in future code edits :P

*a lot nicer
live in the now @kandersolar ! Enjoy the beautiful units. I will make it my personal mission to support your future PRs with units revisions if you don't want to type out the superscript 😂

**Comment by RDaxini** [comment]
Do we need a whatsnew entry for this PR?

**Comment by cwhanse** [comment]
> Do we need a whatsnew entry for this PR? I am guessing not(?)

I think yes, in the Documentation section

File: pvlib/irradiance.py
Comment (echedey-ls) [inline pvlib/irradiance.py:532]:
```suggestion
        Global horizontal irradiance (Wm⁻²).
```

File: pvlib/irradiance.py
Comment (echedey-ls) [inline pvlib/irradiance.py:549]:
```suggestion
        Ground reflected irradiance (Wm⁻²).
```

**Comment by RDaxini** [comment]
@echedey-ls I removed the `:math:` but then added [] rather than () as I think that is more common across other pvlib functions.

**Comment by RDaxini** [comment]
What's up with the pytest command not found error since [39a7ddb](https://github.com/pvlib/pvlib-python/pull/2191/commits/39a7ddb906ec783e3682b4fbda29b7bb98062831)? :S

**Comment by echedey-ls** [comment]
> @echedey-ls I removed the `:math:` but then added [] rather than () as I think that is more common across other pvlib functions.

Yup, I prefer that too. `poa_components` still has parenthesis thou, that's why I thought the next code belonged to it.

> What's up with the pytest command not found error since [39a7ddb](https://github.com/pvlib/pvlib-python/pull/2191/commits/39a7ddb906ec783e3682b4fbda29b7bb98062831)? :S

Once again, witchcraft. I will blind shot that in a PR.

**Comment by RDaxini** [comment]
> Yup, I prefer that too. `poa_components` still has parenthesis thou, that's why I thought the next code belonged to it.

I think the original author might have done that because in that case it looks like the units is part of the sentence rather than just at the end of a parameter name with/without additional info after it that can be read and makes sense in isolation.

I'm sure we'll be able to do a thorough review of units/sentences like this following #2209. I will neaten up that PR and mark it ready for review soon.

File: pvlib/irradiance.py
Comment (kandersolar) [inline pvlib/irradiance.py:798]: I am -0.5 to changing the symbol for surface tilt from $\beta$ to $\theta_T$ .  In the wild west of PV symbols, $\beta$ for surface tilt (and for angles up from the horizon generally) is at least somewhat widespread.  It's also the symbol used in the Loutzenhiser reference.

File: pvlib/irradiance.py
Comment (kandersolar) [inline pvlib/irradiance.py:1533]: How do these changes relate to #2209?  Should we stick with one approach over the other?

Comment (RDaxini) [reply pvlib/irradiance.py:1533]: I got the impression from the discussion in the related issue (#2205) that Wm⁻² was the prefered formatting but the problem was how to type that out, so I figured if #2209 gets merged, since that addresses the inconvenience of typing it out, then we would slowly move towards Wm⁻². That's the change here, although in this case I did just type it out.
I am not opposed to reverting the change for now if that keeps this PR more focused though (following up on the useful advice in one of your other comments)

File: pvlib/irradiance.py
Comment (kandersolar) [inline pvlib/irradiance.py:2155]:
```suggestion
        Global horizontal irradiance from clear sky model. [Wm⁻²]
```

File: pvlib/irradiance.py
Comment (kandersolar) [inline pvlib/irradiance.py:2158]:
```suggestion
        Direct normal irradiance from clear sky model. [Wm⁻²]
```

**Review by kandersolar — APPROVED** [review]

File: docs/sphinx/source/whatsnew/v0.11.1.rst
Comment (RDaxini) [inline docs/sphinx/source/whatsnew/v0.11.1.rst:52]: I noticed this whatsnew entry is missing a link to its PR (#2156) so should I add that in as part of this PR? The tag was also "testing", so I am wondering whether this entry for #2156/#2105 should be moved from documentation to testing. Besides that, I think I have addressed all reviewers' suggestions.

Comment (cwhanse) [reply docs/sphinx/source/whatsnew/v0.11.1.rst:52]: > be moved from documentation to testing

yes but not in this PR

Comment (RDaxini) [reply docs/sphinx/source/whatsnew/v0.11.1.rst:52]: Is it possible to update it in the original merged PR? If not (or not advised) I can just open a new one.

Comment (cwhanse) [reply docs/sphinx/source/whatsnew/v0.11.1.rst:52]: AFAIK it is not possible to edit a merged PR (they can be reverted but not changed). A new PR, or a maintainer can do it directly in the whatsnew file.

Comment (kandersolar) [reply docs/sphinx/source/whatsnew/v0.11.1.rst:52]: We usually just take care of this stuff all at once as part of the "finalize release" PR.  That PR will be editing the whatsnew file anyway (to set the release date if nothing else).  Not worth worrying about cleaning up the whatsnew file until then IMHO.

**Review by cwhanse — APPROVED** [review]
If you care to, I would change the description of `return_components` to

```
If `True`, `diffuse_components` is returned.
```

The markup in the first Returns section could be improved using double `` rather than single `, in places.

**Comment by RDaxini** [comment]
> If you care to, I would change the description of `return_components` to
> 
> ```
> If `True`, ``diffuse_components`` is returned.
> ```

Done

> The markup in the first Returns section could be improved using double `` rather than single `, in places.

I think I have resolved this now, with ` for parameter values and `` for parameters, let me know if I missed anything. If not, I think all issues have now been resolved.

**Comment by cwhanse** [comment]
Changes to the Returns section are great. I was also suggesting that we edit the description of the input parameter `return_components`.

**Comment by RDaxini** [comment]
> Changes to the Returns section are great. I was also suggesting that we edit the description of the input parameter `return_components`.

ohh right of course, I just misread/misunderstood. Check the last commit --- let me know if we're on the same page now 😅

**Review by cwhanse — APPROVED** [review]

**Comment by kandersolar** [comment]
Thanks @RDaxini!

**Comment by RDaxini** [comment]
> Thanks @RDaxini!

and thanks for all the reviews!
btw good idea with the rename 👍🏾

**Pull request merged**
